core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/datalakemetrics/datasizes/GetDatalakeDataSizesHandler.java

```diff
@@ -0,0 +1,159 @@
+import com.sequenceiq.cloudbreak.service.GatewayConfigService;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+import com.sequenceiq.cloudbreak.view.InstanceMetadataView;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+
+@Component
+public class GetDatalakeDataSizesHandler extends ExceptionCatcherEventHandler<GetDatalakeDataSizesRequest> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GetDatalakeDataSizesHandler.class);
+
+    private static final String GET_DATABASE_SIZES_STATE = "datalake_metrics.get_database_sizes.get_database_sizes";
+
+    private static final String GET_SOLR_HBASE_DATA_SIZES_STATE = "datalake_metrics.get_solr_hbase_data_sizes.get_solr_hbase_data_sizes";
+
+    private static final String CORE_INSTANCE_GROUP_NAME = "core";
+
+    private static final String SALT_RESPONSE_STATE_FIELD = "__sls__";
+
+    private static final String SALT_RESPONSE_SUCCESS_FIELD = "result";
+
+    private static final String SALT_RESPONSE_OUTPUT_FIELD = "changes";
+
+    private static final String SALT_RESPONSE_RESULTS_FIELD = "stdout";
+
+    private static final String[] EXPECTED_SIZE_SOURCES = {"database", "hbase", "solr"};
+
+    @Inject
+    private StackDtoService stackDtoService;
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private HostOrchestrator hostOrchestrator;
+
+    @Override
+    public String selector() {
+        return EventSelectorUtil.selector(GetDatalakeDataSizesRequest.class);
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long stackId, Exception ex, Event<GetDatalakeDataSizesRequest> request) {
+        LOGGER.error("Failed to get the datalake data sizes for stack with ID: '" + stackId.toString() + '\'', ex);
+        return new DetermineDatalakeDataSizesFailureEvent(stackId, ex);
+    }
+
+    @Override
+    protected Selectable doAccept(HandlerEvent<GetDatalakeDataSizesRequest> event) {
+        GetDatalakeDataSizesRequest request = event.getData();
+        Long stackId = request.getResourceId();
+        StackDto stackDto = stackDtoService.getById(stackId);
+        GatewayConfig primaryGatewayConfig = gatewayConfigService.getPrimaryGatewayConfig(stackDto);
+
+        try {
+            String databaseSizes = runSaltStateAndGetResult(
+                    primaryGatewayConfig, GET_DATABASE_SIZES_STATE, primaryGatewayConfig.getHostname(), stackId
+            );
+            String solrHBaseSizes = runSaltStateAndGetResult(
+                    primaryGatewayConfig, GET_SOLR_HBASE_DATA_SIZES_STATE, getTargetHostName(stackDto, primaryGatewayConfig),
+                    stackId
+            );
+            String result = '{' + databaseSizes + ',' + solrHBaseSizes + '}';
```

Comment:
Wouldn't this worth it's own object rather than string concatenate?
I see that your are trying to handle the results as a json, but we have libs to do that.

Reply:
So I only use string concatenation here for simplicity since I do not need to manipulate it, and I need to convert it to a separate object later anyway to send it to the DR service.
In fact, [here](https://github.com/hortonworks/cloudbreak/pull/13550/files#diff-66eb079a97ff6871db75e1f16d99100735729ddf693e71505935a47dd2c030beR13) I do use the JSON libs to parse the string and then convert it to the final object. I just saw no reason of doing it earlier.

Let me know if this makes sense and sounds good with you.